main.rb

```diff
@@ -28,8 +33,8 @@ def my_select
     while i < size
       if block_given?
         array.push(self[i]) if yield(self[i])
-      elsif self[i]
-        array.push(self[i])
+      else
+        to_enum
```

Comment:
**`my_select`** identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- Should return an enumerator if no block is given. The way you have handled the task is actually not returning the enumerator. Cause in this current format, it returns the array. so to cover it we have to add return before `to_enum`. And it will do the trick.